### PR TITLE
Created `RawDataContainer`

### PR DESCRIPTION
**Pull request by NachoSoto**
[sc-11079]
Equivalent to https://github.com/RevenueCat/purchases-ios/pull/956.

### CODE REVIEW

**Comment by NachoSoto** [comment]
Still very new to Kotlin, so feedback welcome :)

File: public/src/main/java/com/revenuecat/purchases/CustomerInfo.kt [hunk 1]
```diff
@@ -130,6 +131,10 @@ data class CustomerInfo constructor(
         return entitlements.all[entitlement]?.latestPurchaseDate
     }
 
+    @IgnoredOnParcel
+    override val rawData: JSONObject
+        get() = jsonObject
```
Comment (vegaro) [inline public/src/main/java/com/revenuecat/purchases/CustomerInfo.kt:136]: We are already exposing `jsonObject`, maybe we should deprecate it in favor of `rawData` and make it private in a future release.

File: public/src/main/java/com/revenuecat/purchases/EntitlementInfo.kt [hunk 2]
```diff
@@ -40,7 +46,11 @@ data class EntitlementInfo constructor(
     val unsubscribeDetectedAt: Date?,
     val billingIssueDetectedAt: Date?,
     val ownershipType: OwnershipType,
-) : Parcelable {
+    val jsonObject: JSONObject
+) : Parcelable, RawDataContainer<JSONObject> {
+    @IgnoredOnParcel
```
Comment (vegaro) [inline public/src/main/java/com/revenuecat/purchases/EntitlementInfo.kt:51]: nitpick, add an extra line above this one

File: public/src/main/java/com/revenuecat/purchases/models/RawDataContainer.kt [hunk 3]
```diff
@@ -0,0 +1,12 @@
+package com.revenuecat.purchases.models
+
+/**
+ * A type which exposes its underlying raw data, for debugging purposes or for getting access
+ * to future data while using an older version of the SDK.
+ */
+interface RawDataContainer<Data> {
```
Comment (vegaro) [inline public/src/main/java/com/revenuecat/purchases/models/RawDataContainer.kt:7]: I would do `RawDataContainer<DataType>` instead

File: public/src/main/java/com/revenuecat/purchases/EntitlementInfo.kt [hunk 4]
```diff
@@ -26,6 +31,7 @@ import java.util.Date
  * there is a billing issue. Check the `isActive` property.
  */
 @Parcelize
+@TypeParceler<JSONObject, JSONObjectParceler>()
```
Comment (beylmk) [inline public/src/main/java/com/revenuecat/purchases/EntitlementInfo.kt:34]: do we need this if we also have @IgnoredOnParcel?

Comment (NachoSoto) [reply public/src/main/java/com/revenuecat/purchases/EntitlementInfo.kt:34]: `jsonObject` isn't ignored though.

File: public/src/main/java/com/revenuecat/purchases/EntitlementInfo.kt (hunk 2 above)
Comment (NachoSoto) [reply public/src/main/java/com/revenuecat/purchases/EntitlementInfo.kt:51]: 👍

File: public/src/main/java/com/revenuecat/purchases/models/RawDataContainer.kt (hunk 3 above)
Comment (NachoSoto) [reply public/src/main/java/com/revenuecat/purchases/models/RawDataContainer.kt:7]: 👍

File: public/src/main/java/com/revenuecat/purchases/CustomerInfo.kt (hunk 1 above)
Comment (NachoSoto) [reply public/src/main/java/com/revenuecat/purchases/CustomerInfo.kt:136]: I'll do that!

Comment (NachoSoto) [reply public/src/main/java/com/revenuecat/purchases/CustomerInfo.kt:136]: Doe. Let me know if what I did is correct.

File: public/src/main/java/com/revenuecat/purchases/CustomerInfo.kt [hunk 5]
```diff
@@ -42,13 +43,15 @@ data class CustomerInfo constructor(
     val allExpirationDatesByProduct: Map<String, Date?>,
     val allPurchaseDatesByProduct: Map<String, Date?>,
     val requestDate: Date,
-    val jsonObject: JSONObject,
+    @Deprecated(
+        "Use rawData instead"
```
Comment (vegaro) [inline public/src/main/java/com/revenuecat/purchases/CustomerInfo.kt:47]: This looks good! I would also add the "replaceWith" parameter of the Deprecated annotation so the IDE suggests replacing jsonObject with rawData automatically.

```
@Deprecated(
    message = "Use rawData instead"
    replaceWith = ReplaceWith("rawData")
)
```

File: public/src/main/java/com/revenuecat/purchases/EntitlementInfo.kt [hunk 6]
```diff
@@ -40,7 +46,12 @@ data class EntitlementInfo constructor(
     val unsubscribeDetectedAt: Date?,
     val billingIssueDetectedAt: Date?,
     val ownershipType: OwnershipType,
-) : Parcelable {
+    val jsonObject: JSONObject
```
Comment (vegaro) [inline public/src/main/java/com/revenuecat/purchases/EntitlementInfo.kt:49]: We should also add the `@Deprecated` here

File: public/src/main/java/com/revenuecat/purchases/CustomerInfo.kt (hunk 5 above)
Comment (NachoSoto) [reply public/src/main/java/com/revenuecat/purchases/CustomerInfo.kt:47]: Oh right thanks.

File: public/src/main/java/com/revenuecat/purchases/EntitlementInfo.kt (hunk 6 above)
Comment (NachoSoto) [reply public/src/main/java/com/revenuecat/purchases/EntitlementInfo.kt]: Done, thanks!

**Review by vegaro — APPROVED** [review]

**Pull request merged**